src/Lykke.MatchingEngine.Connector/Services/TcpOrderSocketService.cs

```diff
@@ -31,57 +32,45 @@ internal class TcpOrderSocketService : ITcpClientService, ISocketNotifyer
-                        return;
-                    }
-
-                    var theNewResponse = data as TheNewResponseModel;
-                    if (theNewResponse != null)
-                    {
-                        _logger?.Add($"Response Id: {theNewResponse.Id}");
-                        _newTasksManager.Compliete(theNewResponse.Id, theNewResponse);
-                    }
-
-                    var theMarketOrderResponse = data as MarketOrderResponseModel;
-                    if (theMarketOrderResponse != null)
-                    {
-                        _logger?.Add($"Response Id: {theMarketOrderResponse.Id}");
-                        _marketOrdersTasksManager.Compliete(theMarketOrderResponse.Id, theMarketOrderResponse);
-                    }
-
-                    var multiLimitOrderResponse = data as MeMultiLimitOrderResponseModel;
-                    if (multiLimitOrderResponse != null)
-                    {
-                        _logger?.Add($"Response Id: {multiLimitOrderResponse.Id}");
-                        _multiOrdersTasksManager.Compliete(multiLimitOrderResponse.Id, multiLimitOrderResponse);
-                    }
-                }
-                catch (KeyNotFoundException exception)
-                {
-                    if (_ignoreErrors)
-                    {
-                        _logger?.Add($"Response: {data.ToJson()}");
-                        _logger?.Add(exception.ToString());
-                    }
-                    else
-                    {
-                        throw;
-                    }
-                }
-            });
+        public void HandleDataFromSocket(object data)
+        {
+            try
+            {
+                switch (data)
+                {
+                    case TheResponseModel theResponse:
+                        _logger?.Add($"Response ProcessId: {theResponse.ProcessId}");
+                        _tasksManager.SetResult(theResponse.ProcessId, theResponse);
+                        break;
+                    case TheNewResponseModel theNewResponse:
+                        _logger?.Add($"Response Id: {theNewResponse.Id}");
+                        _newTasksManager.SetResult(theNewResponse.Id, theNewResponse);
+                        break;
+                    case MarketOrderResponseModel theMarketOrderResponse:
+                        _logger?.Add($"Response Id: {theMarketOrderResponse.Id}");
+                        _marketOrdersTasksManager.SetResult(theMarketOrderResponse.Id, theMarketOrderResponse);
+                        break;
+                    case MeMultiLimitOrderResponseModel multiLimitOrderResponse:
+                        _logger?.Add($"Response Id: {multiLimitOrderResponse.Id}");
+                        _multiOrdersTasksManager.SetResult(multiLimitOrderResponse.Id, multiLimitOrderResponse);
+                        break;
+                }

```

Comment:
Any logic for unknown responce? Let's add at least some logging - data.ToJson()

Reply:
Agree